src/plugins/data_source_management/public/components/utils.ts

```diff
@@ -303,6 +304,23 @@ export const handleDataSourceFetchError = (
   });
 };
 
+export const handleDataSourceViewError = (
```

Comment:
So the naming can be a bit confusing: `handleDataSourceViewError()` creates the notification, the `DataSourceViewError` component renders the popover, and `this.handleSwitchDefaultDatasource()` sets the state of the `DataSourceViewError`. Could `handleDataSourceViewError()` at least be renamed to something like `renderDataSourceViewErrorNotification()` to make it distinct from the other two?

Reply:
You are right, let me rethink the naming here, DataSourceView got something different from other component, naming like this might add the complexity to use it, thanks for pointing it out